text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,337 @@
+
+- [Change Logical IDs to Avoid Potential Collisions](https://github.com/aws/aws-cdk/issues/6421)
+
+Similar to aws/aws-cdk#1687, changing logical ID logic introduces a lot of problems for existing users. Whether `LogicalIds 2.0` can be implemented behind a feature flag should be investigated to avoid breaking existing apps.
+
+- [AliasOptions and VersionOptions should not extend EventInvokeConfigOptions](https://github.com/aws/aws-cdk/issues/6966)
+
+More information needed. @eladb
+
+- [Remove Support for Docker Assets with Parameters]()
+
+More information needed. @eladb
+
+# Drawbacks
+
+The main drawback of this staged release strategy is the complexity we will incur within our build tooling to be more version aware. Though doing things like selectively including or excluding deprecated APIs based on vurrent version allows us to phase the release of breaking changes without having to maintain a long lived branch for the upcoming release, the potential for bugs in the functionality of these features could have downsides to users. If a deprecated API is removed from a module when it shouldn't be, and that module is published, it will break user's applications. Though this could of course be remedied in a patch release.
+
+If a deprecated API isn't removed from a version that it was inteded to be, and users start relying on the presence of that API, then by fixing the bug and removing the API in future versions we potentionally introduce breaking changes that weren't intended by us or expected by the users.
+
+Because of this reason, automated verification of our different builds and the presence/absence of APIs and feature flags is, as always, of high importance.
+
+The cost of migration for users varies depending on the state of the application being migrated. All users upgrading to v2 will have to perform the following steps:
+
+1. Update Dependencies - Changing from referencing the old `@aws-cdk/` packages to the new `aws-cdk-lib` package in the corresponding target language's package manifest (package.json, .sln, pom.xml, setup.py).
+2. Update Imports - Since the current module structure is being changed, import statements in user's code will need to change to accommodate the new structure. This will be automated wherever needed with the [import rewriter](#import-rewriter).
+
+Users may have to make additional code changes when upgrading if they are relying on deprecated APIs or feature flags that are being removed. This cost can be minimized through good documentation and information in the [migration guide](#migration-guide).
+
+# Rationale and Alternatives
+
+## Rationale
+
+The main motivation for moving forward with a 2.0 release now is the desire to start moving towards the single module structure. The current multi module structure is known to cause a lot of confusion for users. For more details on why aws-cdk is moving to a single package, see the [MonoCDK RFC](https://github.com/aws/aws-cdk-rfcs/issues/6).
```

Comment:
It's not only user confusion or a "belief" that this is better. It's basically impossible to use CDK modules as peer dependencies since every transitive dependency change will be breaking.

Reply:
Will rethink the wording here. Mostly just trying to say "go read the mono rfc".